ryu/d2s.c

```diff
@@ -548,8 +548,14 @@ void d2s_buffered(double f, char* result) {
   result[index++] = '0' + exp % 10;
 #endif // NO_DIGIT_TABLE
 
+  return index;
```

Comment:
You also need to change the early-return cases for `"Infinity"`/`"NaN"` -- right now the code does `strcpy()` followed by `return;`  Instead you want to do something like:
```
static const char str_nan[] = { 'N', 'a', 'N' };
memcpy(result, str_nan, sizeof(str_nan));
return (int) sizeof(str_nan);
```

Reply:
Oh thanks, didn't notice the other return statements. Will update.

Reply:
Fixed.